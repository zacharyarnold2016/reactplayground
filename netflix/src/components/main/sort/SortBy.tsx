import React, { useState, useCallback } from "react";
import { AiFillCaretDown } from "react-icons/ai";
import { useSelector } from "react-redux";
import { SORTOPTIONS } from "../../../interfaces/enums/lists.enums";
import SortByOption from "./SortByOption";

const SortBy = () => {
  const [open, setOpen] = useState(false);
  const { currentSortString } = useSelector((state: any) => state.film);

  const options = Object.values(SORTOPTIONS);

  const handleOpen = useCallback(() => {
    setOpen(!open);
  }, [open]);
  return (
    <div className="sortBy">
      <h1 style={{ color: "#555555", paddingRight: "5px" }}>Sort By: </h1>
      <div style={{ display: "inline-flex" }}>
        <h1 onClick={handleOpen}>
          {currentSortString} <AiFillCaretDown className="caret" />
        </h1>

        {open && (
          <ul>
            {options.map((element) => {
              return (
                <li key={element.toString()}>
                  <SortByOption option={element} />
                </li>
              );
            })}
          </ul>
        )}
      </div>
    </div>
  );
};
export default SortBy;
