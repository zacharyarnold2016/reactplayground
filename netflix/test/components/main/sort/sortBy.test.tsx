// @ts-nocheck
import { expect } from "@jest/globals";
import renderer from "react-test-renderer";
import SortBy from "../../../../src/components/main/sort/SortBy";
import * as route from "react-router-dom";
import { useState } from "react";
import { useNavigate, useParams } from "react-router";

jest.mock("react-router-dom");
const spyFunction = jest.fn();
useState = jest.fn(() => [1, spyFunction]);
useNavigate = jest.fn(() => spyFunction);

describe("SortBy", () => {
  test("Snapshot Test", () => {
    route.useSearchParams.mockReturnValue([{ get: spyFunction }, spyFunction]);
    const component = renderer.create(<SortBy />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("onClick", () => {
    route.useSearchParams.mockReturnValue([{ get: spyFunction }, spyFunction]);
    const component = renderer.create(<SortBy />);
    let tree = component.toJSON();
    console.log(tree.children[1].children[0].props.onClick());
    expect(spyFunction).toBeCalled();
  });
});
