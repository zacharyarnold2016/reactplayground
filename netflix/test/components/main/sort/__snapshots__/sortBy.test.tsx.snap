// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy Snapshot Test 1`] = `
<div
  className="sortBy"
>
  <h1
    style={
      Object {
        "color": "#555555",
        "paddingRight": "5px",
      }
    }
  >
    Sort By: 
  </h1>
  <div
    style={
      Object {
        "display": "inline-flex",
      }
    }
  >
    <h1
      onClick={[Function]}
    >
       
      <svg
        className="caret"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
        />
      </svg>
    </h1>
    <ul>
      <li>
        <h5
          onClick={[Function]}
          style={
            Object {
              "padding": "1px",
            }
          }
        >
          Release Date
        </h5>
      </li>
      <li>
        <h5
          onClick={[Function]}
          style={
            Object {
              "padding": "1px",
            }
          }
        >
          Vote Average
        </h5>
      </li>
    </ul>
  </div>
</div>
`;
